### ISSUE TITLE
Steam Input broken since 1477

### ISSUE THREAD

**Issue by dragoonDorise** [yuzu-emu/yuzu#10971]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

Mainline 1477 to 1482 / Early Access 3728

### Description of Issue

Steam Input controllers are no recognized since 1477, 1476 is the last one with this working properly

### Expected Behavior

Every controller should appear as an Xbox One Controller

### Reproduction Steps

Install 1476, open from Steam, go to controllers, see that you can select the controller as a Xbox One Controller.
Install 1476, open from Steam, go to controllers, see that you no longer have that option, you'll se the real controller name instead.

### Log File

No need to open a game, just opening the Emulator shows the issue

### System Configuration

CPU: Ryzen 7
GPU/Driver: Nvidia 2600
RAM: 16GB
OS: Windows 10

**Comment by german77** [comment]
We updated to SDL 2.28.0 in 1477. Does the controller work in both if you remap it?

**Comment by Squall-Leonhart** [comment]
> We updated to SDL 2.28.0 in 1477. Does the controller work in both if you remap it?

There was a note that steam input would need to be rebound in a pr leading up to 2.28 iirc.

**Comment by Write** [comment]
Since around EA 3637 I can't use "Any" option and I'm forced to select controller now.

**Comment by german77** [comment]
I tested around with steam input. It doesn't enable the controller until the game starts rendering. This doesn't seem to be yuzu or SDL fault. More like steam input not enabling them.

**Comment by Write** [comment]
Really not sure if it's related to this issue but I can confirm that on 3638, I can connect one controller (trough Steam), disconnect it, and connect an other new one, and both will work instantly. Which allows me to not having to open any settings when switching between two controllers. However, exactly at build 3639, Steam Input simply won't work anymore (not sure if it's exactly Steam Input or something else) and I'm forced to manually select my controller in Settings. Which makes me not able to switch controller seamlessly. It never worked again after that and I'm staying at this build for this feature only. I have my controller settings set on "Any".

**Comment by Write** [comment]
> I was never told you where using two controllers with the same input profile. The behavior you are describing seems like the correct one. However I do see the inconvenience of having to remap after changing your controller. Let me see if there's anything to restore the old behavior.

(Just in case it's not exactly clear I'm not OP)

Thanks a lot !

**Comment by dragoonDorise** [comment]
> I tested around with steam input. It doesn't enable the controller until the game starts rendering. This doesn't seem to be yuzu or SDL fault. More like steam input not enabling them.

If it works on 1476 and not on 1477, it seems maybe the SDL update did this. Can you send me a build with just the old SDL changes so I can test it? I’m a supporter on your Patreon if you need to do that on early

**Comment by german77** [comment]
It's 100% the new SDL update. You can try by replacing 1477 sld2.dll with the one from 1476 on latest mainline versions it won't work due using newer SDL features on yuzu.

The thing is that I'm not going to revert this update. If it's a bug it should be fixed. But if it's a new behavior there's not much I can do. I will try to solve the mapping issue or at the very least know what's actually happening.

**Comment by giodude12** [comment]
I have similar issues just trying to use multiple controllers without having to re-set them up each time. If steam does initialize the xinput controller when the game starts rendering, why does yuzu not recognize this when refreshing controllers?

Also, where can the old sld2.dll be found?

**Comment by Write** [comment]
Thanks a lot guys, indeed restoring old SDL2.dll works flawlessly I'm so happy I can finally use latest Yuzu relaese <3

**Comment by dragoonDorise** [comment]
@german77 is that an accepted workaround? Will it broke yuzu in the future?

**Comment by kevinjalbert** [comment]
Just noting I had the same issue here.

I fixed it by replacing the SDL2.dll with a previous version 2.0.20, this ended up fixing my issue.

This is a post I made in the Yuzu discord, figure it could provide some extra details on the issue (the juice bit is the double `Xbox One Controller 0`, afterwards I now see a 0 and 1): 

> Curious how does the controller input 'any' work?
> I'm trying to get yuzu to work through steam link (host is Win) on my iOS Device.
> https://shottr.cc/s/EoH6/SCR-20230729-rh0.png
> I can see that I have multiple `Xbox One Controller 0` inputs found in Yuzu (One for Xbox Controller, one for iOS Touch Controls). It seems that if I toggle the right one, then everything works.
> If I use a different Steam Link client (Apple TV for example) it doesn't work. I can see the `Xbox One Controller 0`  but it doesn't pick it up as the 'working input' -- If I select it then it will start working.
> At this point, I suspect this is a Steam issue, where the controller inputs from different clients are _different_? Is there anyway to have Yuzu just allow the first 'input' used to be the selected one? I figured that is what `any` would be, but it doesn't seem to behave like that.

**Comment by Write** [comment]
> Just noting I had the same issue here.
> 
> I fixed it by replacing the SDL2.dll with a previous version 2.0.20, this ended up fixing my issue.
> 
> This is a post I made in the Yuzu discord, figure it could provide some extra details on the issue (the juice bit is the double `Xbox One Controller 0`, afterwards I now see a 0 and 1):

Yup, I indeed, had exactly the same issue, with an iOS device too, but basically every devices have the same buggy behavior, and I had to manually select the device every time for it to work. As stated here, reverting to the "old" SDL2.dll fix the issue entierly and makes the option "any" work again correctly.

**Comment by psimona** [comment]
Same issue here, I am forced to select a controller. Using PS5 controller via bluetooth and via steam remote play.

> @german77 is that an accepted workaround? Will it broke yuzu in the future?

Wondering the same, will this bug be adressed?
